File: reachAtNthPoint.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution{
	public:
		int nthPoint(int n){
		    // Code here
		    int prev=1,curr=1,mod=1e9+7,i=2;
		    while(i<=n){
		        int t=curr;
		        curr=(prev+curr)%mod;
		        prev=t;
		        i++;
		    }
		 return curr;
		}
};

//{ Driver Code Starts.
int main(){
	int tc;
	cin >> tc;
	while(tc--){
		int n;
		cin >> n;
		Solution ob;
		int ans  = ob.nthPoint(n);
		cout << ans <<"\n";
	}
	return 0;
}
// } Driver Code Ends

File: reachAtNthPoint.py
# User function Template for python3

import sys


class Solution:
    def nthPoint(self, n):
        # Code here
        curr = prev = 1
        i = 2
        mod = 1e9+7
        while i <= n:
            t = curr
            curr = (curr+prev) % mod
            prev = t
            i += 1
        return int(curr)


# {
 # Driver Code Starts
# Initial Template for Python 3

sys.setrecursionlimit(10**6)

if __name__ == '__main__':
    T = int(input())
    for i in range(T):
        n = int(input())
        ob = Solution()
        ans = ob.nthPoint(n)
        print(ans)
# } Driver Code Ends
